content/en/blog/2023/sig-architecture-conformance-spotlight.md

```diff
@@ -0,0 +1,194 @@
+regarding conformance profiles, exploring the possibility of including
+optional endpoints like RBAC, which are widely used by most end users,
+in specific profiles. However, this aspect is still a work in
+progress.
+
+Endpoints that do not meet the conformance criteria are listed in
+[ineligible_endpoints.yaml](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/ineligible_endpoints.yaml),
+which is publicly accessible in the Kubernetes repo. This file can be
+updated to add or remove endpoints as their status or requirements
+change. These ineligible endpoints are also visible on
+[APISnoop](https://apisnoop.cncf.io/).
+
+Ensuring transparency and incorporating community input regarding the
+eligibility or ineligibility of endpoints is of utmost importance to
+SIG Architecture.
+
+**FSM**: Writing tests for new features is something generally
+requires some kind of enforcement. How do you see the evolution of
+this in Kubernetes? Was there a specific effort to improve the process
+in a way that required tests would be a first-class citizen, or was
+that never an issue?
+
+**RK**: When discussions surrounding the Kubernetes conformance
+programme began in 2018, only approximately 11% of endpoints were
+covered by tests. At that time, the CNCF's governing board requested
+that if funding were to be provided for the work to cover missing
+conformance tests, the Kubernetes Community should adopt a policy of
+not allowing new features to be added unless they include conformance
+tests for their stable APIs.
+
+SIG Architecture is responsible for stewarding this requirement, and
+[APISnoop](https://apisnoop.cncf.io/) has proven to be an invaluable
+tool in this regard. Through automation, APISnoop generates a pull
+request every weekend to highlight any discrepancies in Conformance
+coverage. If any endpoints are promoted to General Availability
+without a conformance test, it will be promptly identified. This
+approach helps prevent the accumulation of new technical debt.
+
+Additionally, there are plans in the near future to create a release
+informing job, which will add an additional layer to prevent any new
+technical debt.
+
+**FSM**: I see, tooling and automation play an important role
+there. What are, in your opinion, the areas that, conformance-wise,
+still require some work to be done? In other words, what are the
+current priority areas marked for improvement?
+
+**RK**: We have reached the “100% Conformance Tested” milestone in
+release 1.27!
+
+At that point, the community took another look at all the endpoints
+that were listed as ineligible for conformance. The list was populated
+through community input over several years.  Several endpoints
+that were previously deemed ineligible for conformance have been
+identified and relocated to a new dedicated list, which is currently
+receiving focused attention for conformance test development. Again,
+that list can also be checked on apisnoop.cncf.io.
```

Comment:
Should we add the URL to this?

Reply:
Here are the URL's
[dedicated list](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/testdata/pending_eligible_endpoints.yaml)

[apisnoop.cncf.io](https://apisnoop.cncf.io/conformance-progress/pending-endpoints)

Reply:
Done.